api/v1beta1/types.go

```diff
@@ -599,6 +601,12 @@ type APIServerLoadBalancer struct {
 	AllowedCIDRs []string `json:"allowedCidrs,omitempty"`
 	// Octavia Provider Used to create load balancer
 	Provider string `json:"provider,omitempty"`
+	// Network defines which network should the load balancer be allocated on.
+	//+optional
+	Network *NetworkFilter `json:"network,omitempty"`
+	// Subnet defines which network(s) should the load balancer be allocated on.
+	//+optional
+	Subnets []*SubnetFilter `json:"subnet,omitempty"`
```

Comment:
Do we ever consider this field to have more than 2 elements? I don't think so, it's just for dual stack use case, so we need kubebuilder constraints and a better description here. Also it's "which subnet(s)" and later on the field json serialization has to be plural.

Reply:
I think the full set is likely:
* MaxLength=2
* optional
* listType=atomic

Please could you also change the type to `[]SubnetFilter`? I don't think we would ever add a nil value to the list, and the list itself already has a natural nil value.

Reply:
Yeah I can totally do that, but regarding the `MaxLength=2` that don't seem right to me. What if we have multiple smaller (e.g. /24s) subnets that the VIP can be assigned from?
Something like:
```
root@octavia1:~# openstack subnet list
+--------------------------------------+-----------------+--------------------------------------+---------------------+
| ID                                   | Name            | Network                              | Subnet              |
+--------------------------------------+-----------------+--------------------------------------+---------------------+
| 340932f1-d742-4762-bd4a-f255dd5609dd | ulab1-ipv4-2    | 1346061d-5bd8-461d-bf43-aa03ed06f2e5 | 10.244.249.0/24     |
| 84e8293d-fffb-4283-b2a7-f8dcda30c3ee | ulab1-ipv4-1    | 1346061d-5bd8-461d-bf43-aa03ed06f2e5 | 10.244.250.0/24     |
| f1c878af-6785-4f02-a551-991e005c8e94 | ulab1-ipv6-1    | 1346061d-5bd8-461d-bf43-aa03ed06f2e5 | 2a02:598:244:1::/64 |
+--------------------------------------+-----------------+--------------------------------------+---------------------+
```

Reply:
Now that I think about it.. you might be right. I mostly don't care which subnet gets selected as far as I get my IP. We will end up in specifying only the network name/id so that subnet with some IP space is automatically selected for us.